### PR TITLE
Changes to create and publish snowflake connector aarch64 wheels

### PR DESCRIPTION
**Pull request by tbbharaj**
This PR addresses the issue to build arm64 wheels. To address this issue, I modified github action workflow to create and publish arm64 wheels using cibuildwheels

Issue: https://github.com/snowflakedb/snowflake-connector-python/issues/808

If there are any questions/comments - I'd be happy fix things as needed.

Thank you
Tanveen

### CODE REVIEW

File: pyproject.toml [hunk 1]
```diff
@@ -6,5 +6,5 @@ requires = [
     "wheel",
     "cython",
     # Must be kept in sync with the `setup_requirements` in `setup.py`
-    'pyarrow>=3.0.0,<3.1.0',
+    'pyarrow>=4.0.1',
```
Comment (xhochy) [inline pyproject.toml:9]: This always needs an upper bound as the ABI breaks with every major release.

**Comment by sfc-gh-mkeller** [comment]
Please create an issue and attach it to this PR

**Comment by tbbharaj** [comment]
> Please create an issue and attach it to this PR

Issue: https://github.com/snowflakedb/snowflake-connector-python/issues/808

**Review by sfc-gh-mkeller — CHANGES_REQUESTED** [review]

File: .github/workflows/build_test.yml [hunk 2]
```diff
@@ -116,6 +116,38 @@ jobs:
           name: linux_py${{ matrix.python-version }}
           path: dist/
 
+  build-aarch64:
+    needs: lint
+    name: Build aarch64 linux-${{matrix.python_tag}}
+    runs-on: ${{ matrix.os }}
```
[the remark this answers is not in the document]
Comment (tbbharaj) [reply .github/workflows/build_test.yml:122]: `os` will get picked from matrix, anyways I updated `matrix`

File: .github/workflows/build_test.yml [hunk 3]
```diff
@@ -116,6 +116,38 @@ jobs:
...
+    strategy:
+      matrix:
+        python_tag: [ "cp37", "cp38", "cp39"]
+        include:
+          - {name: Linux, python-version: '3.7', os: ubuntu-latest}
```
[the remark this answers is not in the document]
Comment (tbbharaj) [reply .github/workflows/build_test.yml]: This works but for convenience I updated this to:
```
matrix:
        python_tag: [ "cp37", "cp38", "cp39"]
        include:
           - name: Linux
           - python-version: '3.7'
           - os: ubuntu-latest
```

File: .github/workflows/build_test.yml [hunk 4]
```diff
@@ -116,6 +116,38 @@ jobs:
...
+    steps:
+      - uses: actions/checkout@v2
+      - name: Setup up QEMU
```
[the remark this answers is not in the document]
Comment (tbbharaj) [reply .github/workflows/build_test.yml:135]: Updated in PR to include 
```
- uses: actions/setup-python@v2
        with:
           python-version: ${{ matrix.python-version }}
```

File: pyproject.toml (hunk 1 above)
Comment (tbbharaj) [reply pyproject.toml]: From the discussion I see `This is blocking our scheduled release next week , we will upgrade OS version on our Mac build machine after release and commit this change again after that`

Does this mean I should wait until OS version is upgraded and after that we can use Arrow 4.x.x?

**Comment by tbbharaj** [comment]
@sfc-gh-mkeller Based on discussion #762, Arrow 4.x.x is not getting accepting. 
From the discussion I see `This is blocking our scheduled release next week , we will upgrade OS version on our Mac build machine after release and commit this change again after that`

Does this mean we should wait until OS version is upgraded and after that we can use Arrow 4.x? Also this issue looks like resolved https://issues.apache.org/jira/browse/ARROW-13108

We need minimum Arrow 4.x to build aarch64 wheels, or can we use Arrow 5.x.x?

**Comment by sfc-gh-mkeller** [comment]
> Does this mean we should wait until OS version is upgraded and after that we can use Arrow 4.x? Also this issue looks like resolved [issues.apache.org/jira/browse/ARROW-13108](https://issues.apache.org/jira/browse/ARROW-13108)

@sfc-gh-abhatnagar I see that SNOW-331072 is done, I'll update our `pyarrow` version to see if the Mac builds work now. @tbbharaj I believe that they might have used a 10.15 specific APIs, so the build might not work properly even on `10.14`.

**Comment by tbbharaj** [comment]
@sfc-gh-mkeller I have resolved the conflicts with `pyarrow` updated to `>=5.0.0,<5.0.1`. Should we be good to merge this PR?

**Comment by tbbharaj** [comment]
@sfc-gh-mkeller Following up on this, if we are good to merge this PR.

**Comment by tbbharaj** [comment]
Checking back on this PR to see if we can add aarch64 wheels support...Thanks!

**Comment by sfc-gh-mkeller** [comment]
@tbbharaj hope you are doing well, I've added the same build job in https://github.com/snowflakedb/snowflake-connector-python/pull/1091 along with M1 wheel builds.
Thank you for your PR, it did help me a lot when I was moving the build jobs to use `cibuildwheel`.

**Pull request closed without merging**